subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/inputs/undeclared/UndeclaredBuildInputsIntegrationTest.groovy

```diff
@@ -592,6 +600,58 @@ class UndeclaredBuildInputsIntegrationTest extends AbstractConfigurationCacheInt
         outputContains("value = value")
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/25044")
+    def "plugin can read file within jar"() {
+        def testFile = JarUtils.jar(testDir.file("thing.jar")) {
+            manifest {}
+
+            entry("foo.txt", "bar")
+        }
+
+        buildKotlinFile << """
+            import ${DefaultTask.name};
+            import ${Project.name};
+            import ${Plugin.name};
+            import ${TaskAction.name};
+            import ${File.name};
+            import ${URI.name};
+            import ${FileSystems.name};
+            import ${Files.name};
+            import ${Path.name};
+
+            public class SomePlugin: Plugin<Project> {
+                override fun apply(project: Project) {
+                    project.tasks.register("readFileWithinJarTask", ReadFileWithinJarTask::class.java)
+                }
+            }
+
+            abstract class ReadFileWithinJarTask : DefaultTask() {
+                @TaskAction
+                fun run() {
+                    try {
+                        val path: Path = File("${testFile.path}").toPath()
```

Comment:
:x: Embedding a file name in a string literal is tricky on Windows because of the `\` which become escape characters. It's better to escape the string:

```suggestion
                        val path: Path = File("${TextUtil.escapeString(testFile.path)}").toPath()
```

Reply:
fixed